frontend/awx/access/credentials/hooks/useCredentialsFilters.tsx

```diff
@@ -1,25 +1,18 @@
-import { useMemo } from 'react';
-import { IToolbarFilter } from '../../../../../framework';
+import { useDynamicToolbarFilters } from '../../../common/useDynamicFilters';
 import {
   useCreatedByToolbarFilter,
-  useDescriptionToolbarFilter,
   useModifiedByToolbarFilter,
-  useNameToolbarFilter,
 } from '../../../common/awx-toolbar-filters';
 
 export function useCredentialsFilters() {
-  const nameToolbarFilter = useNameToolbarFilter();
-  const descriptionToolbarFilter = useDescriptionToolbarFilter();
   const createdByToolbarFilter = useCreatedByToolbarFilter();
   const modifiedByToolbarFilter = useModifiedByToolbarFilter();
-  const toolbarFilters = useMemo<IToolbarFilter[]>(
-    () => [
-      nameToolbarFilter,
-      descriptionToolbarFilter,
-      createdByToolbarFilter,
-      modifiedByToolbarFilter,
-    ],
-    [nameToolbarFilter, descriptionToolbarFilter, createdByToolbarFilter, modifiedByToolbarFilter]
-  );
-  return toolbarFilters;
+  const toolBarFilters = useDynamicToolbarFilters({
+    optionsPath: 'credentials',
+    preSortedKeys: ['name', 'credential_type', 'created-by', 'modified-by'],
+    preFilledValueKeys: ['name', 'credential_type', 'id'],
```

Comment:
Adding `credential_type` to `knownAwxFilterKeys` means we pre-fetch the values - we don't need to add it here.

Reply:
I put it there so that credential type would be one of the first filters in the dropdown as Salma suggested. Is there another way you'd suggest doing that?

Reply:
It doesn't need to be in `preFilledValueKeys`.